### PR TITLE
Migration of build pipeline to GitHub actions

### PR DESCRIPTION
**Pull request by stefancyliax**
﻿Migration of build pipeline to Github Actions from TravisCI

- Pipeline was migrated to Github Actions
- Protobuf updated to current version

### CODE REVIEW

**Comment by stefancyliax** [comment]
@pmai Can you review please?
I forked of you and added 
- deploy to gh-pages
- update to current proto version

File: README.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 Open Simulation Interface (OSI)
 ===============================
 
-[![Travis Build Status](https://travis-ci.com/OpenSimulationInterface/open-simulation-interface.svg?branch=master)](https://travis-ci.com/OpenSimulationInterface/open-simulation-interface)
+[![ProtoBuf CI Builds](https://github.com/OpenSimulationInterface/open-simulation-interface/actions/workflows/protobuf.yml/badge.svg)](https://github.com/OpenSimulationInterface/open-simulation-interface/actions/workflows/protobuf.yml)
```
Comment (pmai) [inline README.md:4]: I don't think those links will work, they should probably be:

```suggestion
[![GitHub Build Status](https://github.com/OpenSimulationInterface/open-simulation-interface/workflows/ProtoBuf%20CI%20Builds/badge.svg)](https://github.com/OpenSimulationInterface/open-simulation-interface/actions?query=workflow%3A%22ProtoBuf+CI+Builds%22)
```

**Review by pmai — CHANGES_REQUESTED** [review]
I don't think those links will work for the badges, it must be keyed off the workflow name.

Other than the screwed up DCO this looks good to me.

File: README.md (hunk 1 above)
Comment (stefancyliax) [reply README.md:4]: This is right out of the Github Generator for Pipeline Badges.
image

**Comment by kmeids** [comment]
**Output CCB 28.04.2021:**
1. To be done after the **3.3.1** release.
2. @pmai  please rebase for DCO
3. @stefancyliax re-do the DCO

**Review by pmai — APPROVED** [review]

**Comment by stefancyliax** [comment]
> 3. @stefancyliax re-do the DCO

We discusses that I will merge this after the DCO is sorted out.

**Comment by stefancyliax** [comment]
@pmai Could you please rebase to fix DCO?

**Comment by kmeids** [comment]
**Output CCB 12.05.2021:**
1. @pmai to do the DCO and merge.

**Pull request merged**